selectrum.el

```diff
@@ -869,11 +879,15 @@ Otherwise just return CANDIDATE."
   (remove-text-properties
    0 (length candidate)
    '(face selectrum-current-candidate) candidate)
-  (apply
-   #'run-hook-with-args
-   'selectrum-candidate-selected-hook
-   candidate selectrum--read-args)
-  (setq selectrum--result (selectrum--get-full candidate))
+  (setq selectrum--result
+        (if (and crm-completion-table
+                 (string-match crm-separator selectrum--previous-input-string))
+            selectrum--previous-input-string
+          (apply
+           #'run-hook-with-args
+           'selectrum-candidate-selected-hook
+           candidate selectrum--read-args)
+          (selectrum--get-full candidate)))
```

Comment:
It seems surprising to me that if you press RET on a candidate, then it's included in the results only sometimes. Do you think it would be a good idea to automatically concat the selected candidate to the list, provided it's not there already?

Reply:
Yes, currently the input defines what candidates are submitted, that is confusing with selectrums model, I changed it so that `RET` now always finishes the input with adding the current selected candidate